BitwardenShared/Core/Vault/Helpers/CipherMatchingHelper.swift

```diff
@@ -1,32 +1,74 @@
 import BitwardenSdk
 import Foundation
 
-/// A helper object to handle filtering a list of ciphers that match a URI.
+// MARK: CipherMatchResult
+
+/// An enum describing the strength that a cipher matches a URI.
 ///
-class CipherMatchingHelper {
-    // MARK: Types
+enum CipherMatchResult {
+    /// The cipher is an exact match for the URI.
+    case exact
+
+    /// The cipher is a close match for the URI.
+    case fuzzy
+
+    /// The cipher doesn't match the URI.
+    case none
+}
 
-    /// An enum describing the strength that a cipher matches a URI.
+// MARK: CipherMatchingHelper
+
+/// A helper to handle filtering ciphers that match a URI.
+///
+protocol CipherMatchingHelper { // sourcery: AutoMockable
+    /// Returns the list of ciphers that match the URI.
     ///
-    enum MatchResult {
-        /// The cipher is an exact match for the URI.
-        case exact
+    /// - Parameters:
+    ///   - uri: The URI used to filter the list of ciphers.
+    ///   - ciphers: The list of ciphers to filter.
+    /// - Returns: The list of ciphers that match the URI.
+    ///
+    @available(*, deprecated) // TODO: PM-24290 remove
```

Comment:
💭  Should we add to the description of PM-24290 ticket a bullet point to remove this? Increases the odds of it not getting forgotten.